### ISSUE TITLE
Add basic support for native AOT

### ISSUE THREAD

**Issue by Evangelink** [microsoft/testfx#2293]
## Summary

Add a first version support of native AOT to collect feedback from users and decide how much invest in this direction. For this first version of support, we should aim at detecting/supporting:
- Test methods
- Parameterized test methods through `[DataRow]` and `[DynamicData]`

**Comment by SymbioticKilla** [comment]
Nice one! I think the main problem is a lack of support by some libraries. I have opened two issues for two libraries, including one from Microsoft, to support NativeAOT. The big problem is Newtonsoft.Json, which is still widely used and will not support NativeAOT. But it’s nice to hear that MSTest is future-oriented. A year ago, I decided to move from xUnit to MSTest in our company project because of the higher activity in the project.
